pkgs/stdenv/generic/default.nix

```diff
@@ -127,6 +127,9 @@ let
         "`stdenv.isArm` is deprecated after 18.03. Please use `stdenv.isAarch32` instead"
         hostPlatform.isAarch32;
 
+      # The derivation's `system` is `buildPlatform.system`.
+      inherit (buildPlatform) system;
```

Comment:
Was this supposed to be `hostPlatform`? I just came across this and thought it seemed like a no-op as is.

Reply:
Yes. It is weird at first but this is a Nix thing. Nix doesn't care why you are just building something, just what (possibly remote) builders can be used. It just cares about the build platform.

Reply:
Isn't that already done earlier in the file?: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/default.nix#L88

Based on some quick tests I did, it looks like this the first assignment to `system` (inside the derivation) is what Nix uses to decide the builder. If you remove that but keep the second assignment (in the attrset override outside the derivation), Nix complains that `system` is not set.

Based on your commit message, I thought you were trying to correctly keep the derivation `system` value as `buildPlatform.system`, but make `stdenv.system` evaluate as `hostPlatform.system`. Right now the second assignment just seems redundant.

Reply:
I tested changing `buildPlatform` to `hostPlatform` in the second assignment and it changed `pkgsCross.armv7l-hf-multiplatform.stdenv.system` from `x86_64-linux` to `armv7l-linux` without rebuilding stdenv.

Reply:
Well, that's weird to me. Can you point to lines on `master` rather than this old PR?

Reply:
On master, the first assignment is: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/default.nix#L88 and the second is: https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/default.nix#L142

Reply:
Ah so I think the whole point of the second must be to overwrite the system from the first without changing the derivation hash. And the only reason to do that is to make it the host platform instead of build platform!

So good catch :) if you could make a PR and give them both some comments, that would be great.